### PR TITLE
Fix: CoreData crash

### PR DESCRIPTION
**Pull request by forgotvas**
Fixed crash with core data

### CODE REVIEW

File: OpenEdX/Data/CoursePersistence.swift [hunk 1]
```diff
@@ -18,32 +18,34 @@ public class CoursePersistence: CoursePersistenceProtocol {
...
+                               hasAccess: $0.hasAccess,
+                               courseStart: $0.courseStart,
+                               courseEnd: $0.courseEnd,
+                               enrollmentStart: $0.enrollmentStart,
+                               enrollmentEnd: $0.enrollmentEnd,
+                               courseID: $0.courseID ?? "",
+                               numPages: Int($0.numPages),
+                               coursesCount: Int($0.courseCount),
+                               progressEarned: 0,
+                               progressPossible: 0)}
```
Comment (rnr) [inline OpenEdX/Data/CoursePersistence.swift:38]: move brace on next line please

**Review by rnr — APPROVED** [review]

File: OpenEdX/Data/CoursePersistence.swift (hunk 1 above)
Comment (forgotvas) [reply OpenEdX/Data/CoursePersistence.swift]: done

File: Core/Core/Network/DownloadManager.swift [hunk 2]
```diff
@@ -389,32 +370,30 @@ public class DownloadManager: DownloadManagerProtocol {
...
-            }
+    private func waitingAll() async {
+        let tasks = await persistence.getDownloadDataTasks()
+        for task in tasks.filter({ $0.state == .inProgress }) {
+            self.persistence.updateDownloadState(
+                id: task.id,
+                state: .waiting,
+                resumeData: nil
+            )
+            self.currentDownloadEventPublisher.send(.added)
```
Comment (saeedbashir) [inline Core/Core/Network/DownloadManager.swift:388]: self is redundant.

File: Core/Core/Network/DownloadManager.swift [hunk 3]
```diff
@@ -389,32 +370,30 @@ public class DownloadManager: DownloadManagerProtocol {
...
         }
+        self.downloadRequest?.cancel()
```
Comment (saeedbashir) [inline Core/Core/Network/DownloadManager.swift:390]: self is redundant.

File: Course/Course/Presentation/Container/CourseContainerViewModel.swift [hunk 4]
```diff
@@ -507,7 +507,9 @@ public class CourseContainerViewModel: BaseCourseViewModel {
                     self.router.dismiss(animated: true)
                 },
                 okTapped: {
-                    self.continueDownload()
+                    Task {
+                        await self.continueDownload()
```
Comment (saeedbashir) [inline Course/Course/Presentation/Container/CourseContainerViewModel.swift:511]: How about using weak reference of self here?

**Review by saeedbashir — APPROVED** [review]
LGTM 👍

**Review by volodymyr-chekyrta — APPROVED** [review]
LGTM

**Pull request merged**